back/engines/commercial/multi_tenancy/config/initializers/apartment.rb

```diff
@@ -152,14 +152,24 @@ def parse_tenant_name(request)
 # https://github.com/rails-on-services/apartment/tree/2.11.0#callbacks
 require 'apartment/adapters/abstract_adapter'
 
-module Apartment
-  module Adapters
-    class AbstractAdapter
-      set_callback :switch, :after do |object|
-        # It's not always correct. E.g. it's `public` tracking errors by sentry-rails gem in controllers.
-        # But still extremely useful for errors in background jobs, rake tasks, and the console.
-        Sentry.set_tags(switched_tenant: ::Tenant.schema_name_to_host(object.current)) if defined?(Sentry)
-      end
-    end
+# Registering +switch+ callbacks
+Apartment::Adapters::AbstractAdapter.set_callback :switch, :before do
+  AppConfiguration.reset_instance
+end
+
+Apartment::Adapters::AbstractAdapter.set_callback :switch, :after do |object|
+  # It's not always correct. E.g. it's `public` tracking errors by sentry-rails gem in controllers.
+  # But still extremely useful for errors in background jobs, rake tasks, and the console.
+  Sentry.set_tags(switched_tenant: Tenant.schema_name_to_host(object.current)) if defined?(Sentry)
+end
+
+# Patching +AbstractAdapter+ to invalidate the +AppConfiguration+ instance when the
+# tenant connection is reset.
```

Comment:
What does "when the tenant connection is reset" mean exactly? I don't understand the scenario this is covering.

Reply:
It's when you call `Apartment::Tenant.reset`. There are two methods to change the tenant in Apartment: `switch` and `reset` (which sets the tenant back to public). I don't think we are calling `reset` explicitly in our codebase, but I thought it would make sense to cover all cases here.

However, after one last check, I realized that `Apartment::Adapters::PostgresqlAdapter` completely overrides `Adapters::AbstractAdapterPatch.reset` (without calling `super`), so it probably has zero effect. To be fixed...